engine/src/main/java/org/camunda/bpm/engine/impl/history/producer/DefaultHistoryEventProducer.java

```diff
@@ -1098,6 +1101,15 @@ protected void initHistoricJobLogEvent(HistoricJobLogEventEntity evt, Job job, H
     String hostName = Context.getCommandContext().getProcessEngineConfiguration().getHostname();
     evt.setHostname(hostName);
 
+    if (jobEntity.getJobHandlerType() != null && jobEntity.getJobHandlerType().equals(HistoryCleanupJobHandler.TYPE)) {
+      String timeToLive = Context.getProcessEngineConfiguration().getHistoryCleanupJobLogTimeToLive();
+      if(timeToLive != null) {
+        Integer timeToLiveDays = ParseUtil.parseHistoryTimeToLive(timeToLive);
+        Date removalTime = DefaultHistoryRemovalTimeProvider.determineRemovalTime(ClockUtil.getCurrentTime(), timeToLiveDays);
```

Comment:
At the beginning of the method, we already call `ClockUtil#getCurrentTime` to set the timestamp. Let's reuse the value and not have a difference of some milliseconds between log creation and removal time.